apps/builder/app/builder/features/ai/commands-bar.tsx

```diff
@@ -0,0 +1,273 @@
+import { z } from "zod";
+import { copywriter, type operations, request } from "@webstudio-is/ai";
+import { createCssEngine } from "@webstudio-is/css-engine";
+import { Button, InputField, Label, Text } from "@webstudio-is/design-system";
+import { SpinnerIcon } from "@webstudio-is/icons";
+import {
+  generateJsxElement,
+  generateJsxChildren,
+  getIndexesWithinAncestors,
+  getStyleRules,
+  idAttribute,
+} from "@webstudio-is/react-sdk";
+import { createScope, findTreeInstanceIds } from "@webstudio-is/sdk";
+import { computed } from "nanostores";
+import { useRef, useState } from "react";
+import { getMapValuesByKeysSet } from "~/shared/array-utils";
+import {
+  breakpointsStore,
+  dataSourcesStore,
+  instancesStore,
+  projectStore,
+  propsStore,
+  registeredComponentMetasStore,
+  selectedInstanceSelectorStore,
+  styleSourceSelectionsStore,
+  stylesStore,
+} from "~/shared/nano-states";
+import { applyOperations, patchTextInstance } from "./apply-operations";
+import { useStore } from "@nanostores/react";
+import { restAi } from "~/shared/router-utils";
+import { RequestParamsSchema } from "~/routes/rest.ai";
+import untruncateJson from "untruncate-json";
+import { traverseTemplate } from "@webstudio-is/jsx-utils";
+
+const handleSubmit = async (
+  event: React.FormEvent<HTMLFormElement>,
+  abortSignal: AbortSignal
+) => {
+  const requestParams = Object.fromEntries(
+    new FormData(event.currentTarget).entries()
```

Comment:
I think .entries() call happens automatically when formdata is passed to fromEntries